schema_based_on_diagram.sql

```diff
@@ -0,0 +1,47 @@
+CREATE DATABASE clinic;
+
+CREATE TABLE patients (
+  id SERIAL NOT NULL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  date_of_birth DATE
+);
+
+CREATE TABLE medical_histories (
+  id SERIAL NOT NULL PRIMARY KEY,
+  admitted_at TIMESTAMP NOT NULL,
+  patient_id INT NOT NULL,
+  status VARCHAR(255) NOT NULL,
+  FOREIGN KEY (patient_id) REFERENCES patients (id)
+);
+
+CREATE TABLE treatments (
+  id SERIAL NOT NULL PRIMARY KEY,
+  type VARCHAR(255) NOT NULL,
+  name VARCHAR(255) NOT NULL
+);
+
+CREATE TABLE invoices (
+  id SERIAL NOT NULL PRIMARY KEY,
+  total_amount DECIMAL NOT NULL,
+  generated_at TIMESTAMP NOT NULL,
+  payed_at TIMESTAMP,
+  medical_history_id INT NOT NULL,
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+CREATE TABLE invoice_items (
+  id SERIAL NOT NULL PRIMARY KEY,
+  unit_price DECIMAL NOT NULL,
+  quantity INT NOT NULL,
```

Comment:
- [ ] it would be better if all the foreign keys have `indexes created` so please try to do that in order to match the requirements 
<img width="395" alt="Screenshot 2022-11-15 at 18 14 35" src="https://user-images.githubusercontent.com/61790587/201984207-dcd8db53-ae47-40ea-9b08-79981cbf80d2.png">